src/NServiceBus.Transport.Amqp/AmqpMessageExtensions.cs

```diff
@@ -1,70 +1,62 @@
 ﻿namespace NServiceBus.Transport.Amqp {
-    using System;
     using System.Collections.Generic;
-    using global::Amqp;
-    using global::Amqp.Framing;
+    using Apache.NMS;
     using NServiceBus.Extensibility;
+    using IMessage = Apache.NMS.IMessage;
 
     public static class AmqpMessageExtensions {
         public static void PopulatePropertiesFromNsbMessage (
-            this Message amqpMessage,
+            this IMessage amqpMessage,
             OutgoingMessage outgoingMessage ) {
 
-            var properties = new Properties {
-                MessageId = outgoingMessage.MessageId
-            };
-
-            properties.ReplyTo = GetNsbHeaderValue (
-                Headers.ReplyToAddress,
-                outgoingMessage.Headers,
-                properties.ReplyTo );
-            properties.CorrelationId = GetNsbHeaderValue (
+            amqpMessage.NMSMessageId = outgoingMessage.MessageId;
+            amqpMessage.NMSCorrelationID = GetNsbHeaderValue (
                 Headers.CorrelationId,
                 outgoingMessage.Headers,
-                properties.CorrelationId );
-
-            amqpMessage.Properties = properties;
+                amqpMessage.NMSCorrelationID );
         }
 
         public static void PopulateApplicationPropertiesFromNsbHeaders (
-            this Message amqpMessage,
+            this IMessage amqpMessage,
             Dictionary<string, string> headers ) {
 
-            amqpMessage.ApplicationProperties = new ApplicationProperties ();
             foreach (var headerPropertyName in headers.Keys) {
-                amqpMessage.ApplicationProperties[headerPropertyName] = headers[headerPropertyName];
+                if (!amqpMessage.Properties.Contains ( headerPropertyName ))
+                    amqpMessage.Properties.SetString ( headerPropertyName,
+                        headers[headerPropertyName] );
```

Comment:
Should be more specific and not just add everything as a string.
Perhaps set known properties with appropriate types and then set the rest as strings.